specutils/core/generic.py

```diff
@@ -0,0 +1,119 @@
+from specutils.spectrum1d import Spectrum1D
+
+import numpy as np
+import logging
+
+from astropy.nddata import NDArithmeticMixin, NDSlicingMixin, NDIOMixin
+import astropy.units as u
+from astropy.wcs import WCS
+
+
+class GenericSpectrum1D(NDIOMixin, NDSlicingMixin, NDArithmeticMixin,
+                        Spectrum1D):
```

Comment:
@mwcraig @crawfordsm @MSeifert04 - do you think this is the correct name for this class, or would you suggest something else? (in relation to NDData)